core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,279 @@
+      return this.location._hashChange;
+    },
+
+    set onhashchange(val) {
+      this.location._hashChange = val;
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+interface UrlParamsType {
+  [param: string]: string
+}
+
+class MockUrlService {
+  getPathname(): string {
+    return '/creator-dashboard';
+  }
+
+  getUrlParams(): UrlParamsType {
+    return {
+      mode: 'create'
+    };
+  }
+}
+
+describe('CreateActivityButtonComponent', () => {
+  let component: CreateActivityButtonComponent;
+  let fixture: ComponentFixture<CreateActivityButtonComponent>;
+  let userService: UserService;
+  let urlService: MockUrlService;
+  let explorationCreationService: ExplorationCreationService;
+  let windowRef: MockWindowRef;
+  let ngbModal: NgbModal;
+
+  beforeEach(waitForAsync(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      declarations: [
+        CreateActivityButtonComponent,
+        TranslatePipe
+      ],
+      providers: [
+        UserService,
+        {
+          provide: UrlService,
+          useClass: MockUrlService
+        },
+        {
+          provide: WindowRef,
+          useValue: windowRef
+        },
+        ExplorationCreationService
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateActivityButtonComponent);
+    component = fixture.componentInstance;
+    userService = TestBed.inject(UserService);
+    urlService = TestBed.inject(UrlService);
+    explorationCreationService = TestBed.inject(ExplorationCreationService);
+    ngbModal = TestBed.inject(NgbModal);
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeDefined();
+  });
+
+  it('should initialize and call initCreationProcess()', fakeAsync(() => {
+    const userServiceSpy = spyOn(userService, 'getUserInfoAsync')
+      // This throws "Argument of type 'Promise<{ isLoggedIn: () => boolean;
+      // canCreateCollections: () => boolean; }>' is not assignable to parameter
+      // of type 'Promise<UserInfo>'.". This is because the actual
+      // 'getUserInfoAsync' returns more properties than required. We need to
+      // suppress this error because we need only "isLoggedIn" and
+      // "canCreateCollections" functions for testing.
+      // @ts-expect-error
+      .and.returnValue(Promise.resolve({
+        isLoggedIn: () => true,
+        canCreateCollections: () => true
+      }));
+    const urlParamsSpy = spyOn(urlService, 'getUrlParams').and.returnValue({
+      mode: 'create'
+    });
+    spyOn(component, 'initCreationProcess');
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(userServiceSpy).toHaveBeenCalled();
+    expect(urlParamsSpy).toHaveBeenCalled();
+    expect(component.initCreationProcess).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(true);
+    expect(component.userIsLoggedIn).toBe(true);
```

Comment:
Do we want a pre-check for these before `ngOnInit` is called?

Reply:
Yes, that would be better. Added pre-checks.